### YamlMime:ManagedReference
items:
- uid: System.Threading.ThreadInterruptedException
  id: ThreadInterruptedException
  children:
  - System.Threading.ThreadInterruptedException.#ctor
  - System.Threading.ThreadInterruptedException.#ctor(System.String)
  - System.Threading.ThreadInterruptedException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  - System.Threading.ThreadInterruptedException.#ctor(System.String,System.Exception)
  langs:
  - csharp
  name: ThreadInterruptedException
  nameWithType: ThreadInterruptedException
  fullName: System.Threading.ThreadInterruptedException
  type: Class
  summary: "Eccezione generata quando un <xref href=&quot;System.Threading.Thread&quot;> </xref> viene interrotto mentre è in stato di attesa."
  remarks: "Dopo aver creato un thread, è in uno o più <xref:System.Threading.ThreadState>indica fino a quando non viene eliminato.</xref:System.Threading.ThreadState> La chiamata <xref:System.Threading.Thread.Interrupt%2A>quando un thread si trova il <xref:System.Threading.ThreadState>stato causerà una ThreadInterruptedException un&quot;eccezione nel thread di destinazione.</xref:System.Threading.ThreadState> </xref:System.Threading.Thread.Interrupt%2A> Se il thread non è nel <xref:System.Threading.ThreadState>stato, non viene generata l&quot;eccezione fino a quando il thread di tale stato.</xref:System.Threading.ThreadState> Se il thread si blocca mai, verrà completato senza subire alcuna interruzione.       ThreadInterruptedException utilizza HRESULT COR_E_THREADINTERRUPTED il cui valore è 0x80131519.       Per un elenco di valori di proprietà iniziali per un&quot;istanza di ThreadInterruptedException, vedere il <xref:System.Threading.ThreadInterruptedException.%23ctor%2A>costruttori.</xref:System.Threading.ThreadInterruptedException.%23ctor%2A>"
  example:
  - "The following code example shows the behavior of a running thread when it is interrupted and subsequently gets blocked.  \n  \n [!code-cpp[System.Threading.Thread.Interrupt#1](~/add/codesnippet/cpp/t-system.threading.threa_1_1.cpp)]\n [!code-vb[System.Threading.Thread.Interrupt#1](~/add/codesnippet/visualbasic/t-system.threading.threa_1_1.vb)]\n [!code-cs[System.Threading.Thread.Interrupt#1](~/add/codesnippet/csharp/t-system.threading.threa_1_1.cs)]"
  syntax:
    content: >-
      [System.Runtime.InteropServices.ComVisible(true)]

      public class ThreadInterruptedException : SystemException
  inheritance:
  - System.Exception
  - System.SystemException
  implements: []
  inheritedMembers: []
  platform:
  - net462
- uid: System.Threading.ThreadInterruptedException.#ctor
  id: '#ctor'
  parent: System.Threading.ThreadInterruptedException
  langs:
  - csharp
  name: ThreadInterruptedException()
  nameWithType: ThreadInterruptedException.ThreadInterruptedException()
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException()
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "Inizializza una nuova istanza di <xref href=&quot;System.Threading.ThreadInterruptedException&quot;> </xref> classe con le proprietà predefinite."
  remarks: "Nella tabella seguente vengono illustrati i valori di proprietà iniziali per un&quot;istanza di <xref:System.Threading.ThreadInterruptedException>.</xref:System.Threading.ThreadInterruptedException>      | Proprietà | Valore |   |--------------|-----------|   | <xref:System.Exception.InnerException%2A>| `null`.|   | <xref:System.Exception.Message%2A>| La stringa di messaggio di errore localizzato. |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public ThreadInterruptedException ();
    parameters: []
  overload: System.Threading.ThreadInterruptedException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Threading.ThreadInterruptedException.#ctor(System.String)
  id: '#ctor(System.String)'
  parent: System.Threading.ThreadInterruptedException
  langs:
  - csharp
  name: ThreadInterruptedException(String)
  nameWithType: ThreadInterruptedException.ThreadInterruptedException(String)
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException(String)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "Inizializza una nuova istanza di <xref href=&quot;System.Threading.ThreadInterruptedException&quot;> </xref> classe con un messaggio di errore specificato."
  remarks: "Nella tabella seguente vengono illustrati i valori di proprietà iniziali per un&quot;istanza di <xref:System.Threading.ThreadInterruptedException>.</xref:System.Threading.ThreadInterruptedException>      | Proprietà | Valore |   |--------------|-----------|   | <xref:System.Exception.InnerException%2A>| `null`.|   | <xref:System.Exception.Message%2A>| La stringa di messaggio di errore. |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public ThreadInterruptedException (string message);
    parameters:
    - id: message
      type: System.String
      description: "Il messaggio di errore che spiega il motivo dell&quot;eccezione."
  overload: System.Threading.ThreadInterruptedException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Threading.ThreadInterruptedException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  id: '#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)'
  parent: System.Threading.ThreadInterruptedException
  langs:
  - csharp
  name: ThreadInterruptedException(SerializationInfo,StreamingContext)
  nameWithType: ThreadInterruptedException.ThreadInterruptedException(SerializationInfo,StreamingContext)
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException(SerializationInfo,StreamingContext)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "Inizializza una nuova istanza di <xref href=&quot;System.Threading.ThreadInterruptedException&quot;> </xref> classe con dati serializzati."
  remarks: "Questo costruttore viene chiamato durante la deserializzazione per ricostituire l&quot;oggetto eccezione trasmesso su un flusso. Per ulteriori informazioni, vedere [serializzazione XML e SOAP](~/add/includes/ajax-current-ext-md.md)."
  syntax:
    content: protected ThreadInterruptedException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);
    parameters:
    - id: info
      type: System.Runtime.Serialization.SerializationInfo
      description: "Il <xref href=&quot;System.Runtime.Serialization.SerializationInfo&quot;> </xref> che contiene i dati oggetto serializzati relativi all&quot;eccezione generata."
    - id: context
      type: System.Runtime.Serialization.StreamingContext
      description: "Il <xref:System.Runtime.Serialization.StreamingContext>che contiene le informazioni contestuali sull&quot;origine o destinazione.</xref:System.Runtime.Serialization.StreamingContext>"
  overload: System.Threading.ThreadInterruptedException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Threading.ThreadInterruptedException.#ctor(System.String,System.Exception)
  id: '#ctor(System.String,System.Exception)'
  parent: System.Threading.ThreadInterruptedException
  langs:
  - csharp
  name: ThreadInterruptedException(String,Exception)
  nameWithType: ThreadInterruptedException.ThreadInterruptedException(String,Exception)
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException(String,Exception)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "Inizializza una nuova istanza di <xref href=&quot;System.Threading.ThreadInterruptedException&quot;> </xref> classe con un messaggio di errore specificato e un riferimento all&quot;eccezione interna che è la causa dell&quot;eccezione corrente."
  remarks: "Un&quot;eccezione generata come risultato diretto di un&quot;eccezione precedente deve includere un riferimento all&quot;eccezione precedente nella <xref:System.Exception.InnerException%2A>proprietà.</xref:System.Exception.InnerException%2A> Il <xref:System.Exception.InnerException%2A>proprietà restituisce lo stesso valore passato al costruttore o `null` se il <xref:System.Exception.InnerException%2A>proprietà non fornisce il valore di eccezione interna al costruttore.</xref:System.Exception.InnerException%2A> </xref:System.Exception.InnerException%2A>       Nella tabella seguente vengono illustrati i valori di proprietà iniziali per un&quot;istanza di <xref:System.Threading.ThreadInterruptedException>.</xref:System.Threading.ThreadInterruptedException>      | Proprietà | Valore |   |--------------|-----------|   | <xref:System.Exception.InnerException%2A>| Il riferimento all&quot;eccezione interna. |   | <xref:System.Exception.Message%2A>| La stringa di messaggio di errore. |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public ThreadInterruptedException (string message, Exception innerException);
    parameters:
    - id: message
      type: System.String
      description: "Il messaggio di errore che spiega il motivo dell&quot;eccezione."
    - id: innerException
      type: System.Exception
      description: "Eccezione che è la causa dell&quot;eccezione corrente. Se il `innerException` parametro non è <xref uid=&quot;langword_csharp_null&quot; name=&quot;null&quot; href=&quot;&quot;> </xref>, l&quot;eccezione corrente viene generata in un <xref uid=&quot;langword_csharp_catch&quot; name=&quot;catch&quot; href=&quot;&quot;> </xref> blocco che gestisce l&quot;eccezione interna."
  overload: System.Threading.ThreadInterruptedException.#ctor*
  exceptions: []
  platform:
  - net462
references:
- uid: System.SystemException
  isExternal: false
  name: System.SystemException
- uid: System.Threading.ThreadInterruptedException.#ctor
  parent: System.Threading.ThreadInterruptedException
  isExternal: false
  name: ThreadInterruptedException()
  nameWithType: ThreadInterruptedException.ThreadInterruptedException()
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException()
- uid: System.Threading.ThreadInterruptedException.#ctor(System.String)
  parent: System.Threading.ThreadInterruptedException
  isExternal: false
  name: ThreadInterruptedException(String)
  nameWithType: ThreadInterruptedException.ThreadInterruptedException(String)
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException(String)
- uid: System.String
  parent: System
  isExternal: true
  name: String
  nameWithType: String
  fullName: System.String
- uid: System.Threading.ThreadInterruptedException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  parent: System.Threading.ThreadInterruptedException
  isExternal: false
  name: ThreadInterruptedException(SerializationInfo,StreamingContext)
  nameWithType: ThreadInterruptedException.ThreadInterruptedException(SerializationInfo,StreamingContext)
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException(SerializationInfo,StreamingContext)
- uid: System.Runtime.Serialization.SerializationInfo
  parent: System.Runtime.Serialization
  isExternal: false
  name: SerializationInfo
  nameWithType: SerializationInfo
  fullName: System.Runtime.Serialization.SerializationInfo
- uid: System.Runtime.Serialization.StreamingContext
  parent: System.Runtime.Serialization
  isExternal: true
  name: StreamingContext
  nameWithType: StreamingContext
  fullName: System.Runtime.Serialization.StreamingContext
- uid: System.Threading.ThreadInterruptedException.#ctor(System.String,System.Exception)
  parent: System.Threading.ThreadInterruptedException
  isExternal: false
  name: ThreadInterruptedException(String,Exception)
  nameWithType: ThreadInterruptedException.ThreadInterruptedException(String,Exception)
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException(String,Exception)
- uid: System.Exception
  parent: System
  isExternal: true
  name: Exception
  nameWithType: Exception
  fullName: System.Exception
- uid: System.Threading.ThreadInterruptedException.#ctor*
  parent: System.Threading.ThreadInterruptedException
  isExternal: false
  name: ThreadInterruptedException
  nameWithType: ThreadInterruptedException.ThreadInterruptedException
